controllers/machinedeletionremediation_controller.go

```diff
@@ -76,29 +93,56 @@ type MachineDeletionRemediationReconciler struct {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
-func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (finalResult ctrl.Result, finalErr error) {
 	log := r.Log.WithValues("machinedeletionremediation", req.NamespacedName)
 
 	log.Info("reconciling...")
 
 	var err error
 	var remediation *v1alpha1.MachineDeletionRemediation
 	if remediation, err = r.getRemediation(ctx, req); remediation == nil || err != nil {
-		if apiErrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
 		return ctrl.Result{}, err
 	}
 
 	log.Info("Machine Deletion Remediation CR found", "name", remediation.GetName())
 
+	defer func() {
+		if updateErr := r.updateStatus(ctx, remediation); updateErr != nil {
+			if !apiErrors.IsConflict(updateErr) {
+				finalErr = utilerrors.NewAggregate([]error{updateErr, finalErr})
+			}
+			finalResult.RequeueAfter = time.Second
+		}
+	}()
+
 	// Remediation's name was created from Node's name
 	nodeName := remediation.GetName()
 
-	var machine *unstructured.Unstructured
-	if machine, err = r.getMachine(remediation); err != nil {
+	if r.isStoppedByNHC(remediation) {
```

Comment:
nit, won't block on it: I would prefer consistent wording. NHC puts a `timeout` annotation on the CR. But here it is about `stopped`, below it's `terminated`....